aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -9,10 +9,28 @@
 #include <ATen/NestedTensorImpl.h>
 #include <c10/core/DispatchKey.h>
 #include <ATen/native/nested/NestedTensorMath.h>
+#include "ATen/ops/nested_tensor_native.h"
+#include "c10/core/TensorImpl.h"
 
 namespace at {
 namespace native {
 
+int64_t NestedTensor_size_int(const Tensor& self, int64_t d) {
+  const auto* nt_indices = get_nested_tensor_impl(self);
+  const auto opt_size_d = nt_indices->opt_size(d);
+  TORCH_CHECK(opt_size_d, "Irregular dimension ", d, " does not have a size.");
+  return *opt_size_d;
+}
+
+at::Tensor NestedTensor_nested_size_tensor(const Tensor& self){
+  const auto* nt_indices = get_nested_tensor_impl(self);
+  return nt_indices->get_nested_size_tensor().clone();
```

Comment:
was erroring on t.storage.result.count() without cloning .. feels like something I swept under the rug

Reply:
Another way to get around this is to add to the list: `DONT_ENFORCE_SAME_TENSOR_IMPL_OR_STORAGE` in `tools/autograd/gen_variable_type.py`. This would avoid the extra copy from clone.